### DISCUSSION TITLE
How to integrate Firebase with Tauri?

### DISCUSSION THREAD

**Discussion by elanzini** [tauri-apps/tauri#4805] [Q&A]
I am trying to use Firebase to sign-in users in my app. I followed all the steps in Firebase Getting Started page and was able to get it to work locally with the [redirect-mode](https://firebase.google.com/docs/auth/web/google-signin#redirect-mode). Everything works fine when I am running the app with `npm run tauri dev` but when I build the app and install it (`npm run tauri build`), I am getting `FirebaseError: Firebase: Error (auth/unauthorized-domain).`

Everything online points to the `Authorized domains` settings in Firebase Console but I can't quite understand how everything works fine in `dev` but breaks when the app is installed.

Any ideas?

**Comment by FabianLars** [comment]
Which domains did you allow in `Authorized domains`?

The main difference between `dev` and `build` is that `dev` uses `http://localhost:port` and `build` uses `tauri://localhost/` (macos/linux) and `https://tauri.localhost/` (windows) in a built app.

**Reply by elanzini** [reply]
I have `localhost` in my authorized domains. I tried adding `tauri://localhost/` but it's not allowed in Firebase.
I also tried using the [tauri-invoke-http](https://github.com/tauri-apps/tauri-invoke-http) plugin without luck

<img width="624" alt="Screen Shot 2022-07-31 at 14 31 25" src="https://user-images.githubusercontent.com/37532050/182028737-75fdf646-d632-4336-8b26-e99b933ff8cf.png">

**Reply by elanzini** [reply]
I was able to add `https://tauri.localhost/` to the allowed domains but it looks like that didn't change anything.

**Comment by liyasthomas** [comment]
I had the same problem, how did you solve it? @elanzini 🙏

**Comment by ZivFuchs** [comment]
I had this same issue. I was able to use @FabianLars's [oauth-plugin](https://github.com/FabianLars/tauri-plugin-oauth) to get authentication working in dev mode, but not in the built application. My investigations found that the localhost redirect server was properly started and the auth token was correctly passed to the client, but when I called the Firebase SDK's `signInWithCredential()` function, it seemed to not do anything, and also didn't throw any errors that could be caught with `.catch()`. I also suspect this is related to authorized domains, and the fact that was already pointed out that the domains used by the built Tauri app can't be authorized in the Firebase console.

**Comment by Jun-Murakami** [comment]
I am also facing the same problem.
Firebase rejects referrers for Tauri's production build.
There seems to be no easy solution.
----
ErroCode:auth/requests-from-referer-https://tauri.localhost-are-blocked.
Firebase: Error (auth/requests-from-referer-https://tauri.localhost-are-blocked.).h

**Comment by xkjay** [comment]
I found this:
https://res.cloudinary.com/practicaldev/image/fetch/s--tfMea56H--/c_limit%2Cf_auto%2Cfl_progressive%2Cq_auto%2Cw_880/https://dev-to-uploads.s3.amazonaws.com/uploads/articles/xpfy67mxm4oa91lihkpp.png
It shows the entire authentication flow

From: 
https://marcchen.hashnode.dev/building-desktop-applications-with-tauri-nextjs-firebase

**Comment by Pandananana** [comment]
I have the same issue. I implemented the Tauri HTTP fetch solution from [here](https://marcchen.hashnode.dev/building-desktop-applications-with-tauri-nextjs-firebase) and it returns an Auth token I can use to interact with Firebase, but only through REST API with the Auth token in the HTTP header. I still haven't gotten the Firebase SDK Authenticated correctly so that I easily can use their functions.

**Comment by Pandananana** [comment]
I got it to work! I used initializeAuth() instead of getAuth() and then there was no errors

**Reply by anuj1072001** [reply]
i also used this it works

**Comment by liyasthomas** [comment]
Has anyone successfully integrated Tauri with Firebase OAuth?
